### DISCUSSION TITLE
Issue with deprepro, Error occurred, Exception: EmptyInlineError,Message: Empty inline expression

### DISCUSSION THREAD

**Discussion by ramic-k** [snl-dakota/dakota#24] [Q&A] [answered]
Hello, I am having some issues running dakota with input files that worked flawlessly before.

The error I get is the following:

 `(base) ykr@W132089:~/Desktop/new_PE_dakota_optimization$ dakota -i dakota_coliny_direct_ncrystal_coherent_with_extinction.in -o dakota_coherent_with_extinction.out
Error occurred
    Exception: EmptyInlineError
    Message: Empty inline expression
python: can't open file '/home/ykr/Desktop/new_PE_dakota_optimization/workdir_coherent_with_extinction_from_dft_without_pok/run.1/ncrystal_chi_square_calculate_coliny_direct_coherent_with_extinction.py': [Errno 2] No such file or directory
Error(s) encountered reading results file workdir_coherent_with_extinction_from_dft_without_pok/run.1/ncrystal_results.out for Evaluation 1:
-- Expected 1 function value(s) but found 0.
`

It seems that dprepro didnt sub in the parameters into my tamplete file. This worked before, I didnt play with dakota for few months, and now I encounter this issue. I get the same error with 6.16 and 6.17 version of dakota.

My dakota input file looks like this:

>  \# Dakota PE total cross section optimization with NCrystal
> 
> environment
>   tabular_data
>     tabular_data_file 'coliny_direct_ncrystal_tabular_coherent_with_extinction_without_pok.dat'
> 
> method
>   coliny_direct
>   max_function_evaluations = 100000
>   max_iterations = 2000 
>   min_boxsize_limit = 1e-5
> 
> variables
>   continuous_design 10
>     descriptors    'w1' 'w2' 'w3' 'w4' 'w5' 'w6' 'w7' 'w8' 'w9' 'w10'
>     upper_bounds   10 10 10 10 10 10 10 10 10 10
>     lower_bounds   0 0 0 0 0 0 0 0 0 0
>     initial_point  1 1 1 1 1 1 1 1 1 1
> 
> interface
>   analysis_drivers 'ncrystal_driver_optimize_coliny_direct_coherent_with_extinction.sh'                         # For Linux/Mac
>     fork
>       work_directory named 'workdir_coherent_with_extinction_from_dft_without_pok/run'
>       directory_tag directory_save
>       link_files 'ncrystal_chi_square_calculate_coliny_direct_coherent_with_extinction.template' 'PE_from_dft_original.ncmat' 'ETT_ETTC_PE_AVE_XS.txt' 'VIS_19509-19509_polyethylene_15sheets_0p1mm_each_at_5K.dat' 'c-12-absorption.dat' 'c-13-absorption.dat' 'h-1-absorption.dat'
>       parameters_file 'ncrystal_params.in'
>       results_file 'ncrystal_results.out'
>     deactivate active_set_vector
> 
> responses
>   descriptors 'chisquare_value'
>   objective_functions 1
>   no_gradients
>   #numerical_gradients
>   no_hessians

and my driver looks like this:

> \#!/bin/bash
> 
> \# $1 and $2 are special variables in bash that contain the 1st and 2nd 
> \# command line arguments to the script, which are the names of the
> \# Dakota parameters and results files, respectively.
> 
> params=$1
> results=$2
> 
>#gedit $params
> 
> \############################################################################### 
> \##
> \## Pre-processing Phase -- Generate/configure an input file for your simulation 
> \##  by substiting in parameter values from the Dakota paramters file.
> \##
> \###############################################################################
> 
> dprepro \$params ncrystal_chi_square_calculate_coliny_direct_coherent_with_extinction.template ncrystal_chi_square_calculate_coliny_direct_coherent_with_extinction.py
> 
> \############################################################################### 
> \##
> \## Execution Phase -- Run your simulation
> \##
> \###############################################################################
> 
> chisquare_value=\$( python ncrystal_chi_square_calculate_coliny_direct_coherent_with_extinction.py | tail -1)
> 
> echo \"\$chisquare_value\" > \$results

I have verified that the params file is created and it looks like this:

The params file looks like this:

>                                          10 variables
>                       5.000000000000000e+00 w1
>                       5.000000000000000e+00 w2
>                       5.000000000000000e+00 w3
>                       5.000000000000000e+00 w4
>                       5.000000000000000e+00 w5
>                       5.000000000000000e+00 w6
>                       5.000000000000000e+00 w7
>                       5.000000000000000e+00 w8
>                       5.000000000000000e+00 w9
>                       5.000000000000000e+00 w10
>                                           1 functions
>                                           1 ASV_1:chisquare_value
>                                          10 derivative_variables
>                                           1 DVV_1:w1
>                                           2 DVV_2:w2
>                                           3 DVV_3:w3
>                                           4 DVV_4:w4
>                                           5 DVV_5:w5
>                                           6 DVV_6:w6
>                                           7 DVV_7:w7
>                                           8 DVV_8:w8
>                                           9 DVV_9:w9
>                                          10 DVV_10:w10
>                                           0 analysis_components
>                                           1 eval_id
>                                           0 metadata

The relevant section of my template file looks like this:

> new_rho=np.concatenate((H_PE_rho_initial_values_imported[:index0]\*{w1},H_PE_rho_initial_values_imported[index0:index1]\*{w2},H_PE_rho_initial_values_imported[index1:index2]\*{w3},H_PE_rho_initial_values_imported[index2:index3]\*{w4},H_PE_rho_initial_values_imported[index3:index4],H_PE_rho_initial_values_imported[index4:index5]\*{w5},H_PE_rho_initial_values_imported[index5:index6]\*{w6},H_PE_rho_initial_values_imported[index6:index7]\*{w7},H_PE_rho_initial_values_imported[index7:index8]\*{w8},H_PE_rho_initial_values_imported[index8:index9]\*{w9},H_PE_rho_initial_values_imported[index9:index10],H_PE_rho_initial_values_imported[index10:]\*{w10}))

Your help would be great appreciated.

**Comment by jadamstephens** [comment, marked as answer]
I can't see any obvious problems with what you've sent.

Do you see the problem if you run dprepro outside of the script on the command line?

What if you paste just the snippet of the template into a file and run dprepro on it? If that works, it suggests that the problem is elsewhere in your template.

**Reply by ramic-k** [reply]
Thank you for your quick reply Adam. I made some changes to my template, so it must be something there. 
I ran another old set of dakota files, and that worked fine, so it must be something in my template. I will try to figure out what it is. Thank you for the help!
